### PR TITLE
[MM-43199] api4/elasticsearch: test elasticsearch after checking the config values

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
Add nil value check to elasticsearch settings.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-43199
#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: model/config.go [hunk 1]
```diff
@@ -2459,28 +2459,27 @@ func (s *NativeAppSettings) SetDefaults() {
...
-	PostIndexShards               *int    `access:"environment_elasticsearch,write_restrictable,cloud_restrictable"`
-	ChannelIndexReplicas          *int    `access:"environment_elasticsearch,write_restrictable,cloud_restrictable"`
-	ChannelIndexShards            *int    `access:"environment_elasticsearch,write_restrictable,cloud_restrictable"`
-	UserIndexReplicas             *int    `access:"environment_elasticsearch,write_restrictable,cloud_restrictable"`
-	UserIndexShards               *int    `access:"environment_elasticsearch,write_restrictable,cloud_restrictable"`
-	AggregatePostsAfterDays       *int    `access:"environment_elasticsearch,write_restrictable,cloud_restrictable"` // telemetry: none
-	PostsAggregatorJobStartTime   *string `access:"environment_elasticsearch,write_restrictable,cloud_restrictable"` // telemetry: none
-	IndexPrefix                   *string `access:"environment_elasticsearch,write_restrictable,cloud_restrictable"`
-	LiveIndexingBatchSize         *int    `access:"environment_elasticsearch,write_restrictable,cloud_restrictable"`
-	BulkIndexingTimeWindowSeconds *int    `json:",omitempty"` // telemetry: none
```
Comment (agnivade) [inline model/config.go:2479]: Hmm why is this removed?

Comment (isacikgoz) [reply model/config.go:2479]: It was not being used so decided to remove it because we don't set a default value for this hence breaks the nil checker. Didn't want to add an exception for this one.

Comment (agnivade) [reply model/config.go:2479]: This was set to omitempty during my elasticsearch changes. But we cannot yet remove this field because of compatiblity requirements. Mobile and other plugins will start to fail if we remove any field from the model package. I'm afraid we have to add an exception till the next breaking change.

**Review by agnivade — APPROVED** [review]
